server/middlewares/authorizaton.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable consistent-return */
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import Util from '../helpers/util';
+import userService from '../services/userService';
+import managementService from '../services/userManagementService';
+
+
+dotenv.config();
+
+const util = new Util();
+
+
+export const authorizationCheck = async(req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+    const decoded = jwt.verify(token, process.env.JWT_KEY);
+    const viewToken = await userService.findByProp({ token });
+    if(!viewToken[0]){
+      const Error = 'Unauthorised Access: You have to login to Proceed';
+      util.setError(401, Error);
+      return util.send(res);
+    }
+    req.userData = decoded;
+    next();
+  }catch(error) {
+    const Error = 'No token provided or Token expired';
+    util.setError(404, Error);
+    return util.send(res);
+  }
+};
+
+export const findManagerByUserID = async(req, res, next)=>{
```

Comment:
@irfiacre I think here findManagerByUserId does not correspond to authorization, can u use a descriptive name for that function.

Reply:
`findManagerByUserId ` this a check related to authorizing the to use our endpoint, but is completely independent of the ***`TOKEN VERIFICATION`***   hence I only use that name in order to not be confused with `authorizationCheck` (that is responsible for the token verification).

That's why I included it in the `authorization file` and called it that name.